node-api/backend/validator.go

```diff
@@ -21,80 +21,224 @@
 package backend
 
 import (
+	"slices"
+	"strconv"
+
+	"github.com/berachain/beacon-kit/consensus-types/types"
+	"github.com/berachain/beacon-kit/errors"
 	"github.com/berachain/beacon-kit/node-api/backend/utils"
 	beacontypes "github.com/berachain/beacon-kit/node-api/handlers/beacon/types"
+	"github.com/berachain/beacon-kit/primitives/crypto"
 	"github.com/berachain/beacon-kit/primitives/math"
+	statedb "github.com/berachain/beacon-kit/state-transition/core/state"
 )
 
+type validatorFilters struct {
+	numericIDs []uint64
+	pubkeys    []crypto.BLSPubkey
+}
+
+// parseValidatorIDs parses a slice of string IDs into numeric IDs and pubkeys
+func parseValidatorIDs(ids []string) *validatorFilters {
+	filters := &validatorFilters{
+		numericIDs: make([]uint64, 0, len(ids)),
+		pubkeys:    make([]crypto.BLSPubkey, 0, len(ids)),
+	}
+
+	for _, id := range ids {
+		filters.parseID(id)
+	}
+
+	return filters
+}
+
+// parseID attempts to parse a single ID as either a numeric ID or pubkey
+func (f *validatorFilters) parseID(id string) {
+	// Try parsing as numeric ID first
+	if index, err := strconv.ParseUint(id, 10, 64); err == nil {
+		f.numericIDs = append(f.numericIDs, index)
+		return
+	}
+
+	// Try parsing as pubkey
+	var pubkey crypto.BLSPubkey
+	if err := pubkey.UnmarshalText([]byte(id)); err == nil {
+		f.pubkeys = append(f.pubkeys, pubkey)
+	}
+	// Silently skip invalid IDs
```

Comment:
q: is this correct? Down the line we threat an empty filter as `show all validators`. Should we treat a list of malformed IDs into a catchall?
Consider erring here.
cc @calbera

Reply:
Am indifferent here, really depends on what the spec enforces